h5py/tests/common.py

```diff
@@ -165,3 +167,32 @@ def closed_tempfile(suffix='', text=None):
             test_file.flush()
     yield file_name
     shutil.rmtree(file_name, ignore_errors=True)
+
+
+def insubprocess(f):
+    """Runs a test in its own subprocess"""
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        curr_test = os.environ.get('PYTEST_CURRENT_TEST', None)
+        if not curr_test:
+            raise ut.SkipTest("not running in pytest")
+        curr_test, _, _ = curr_test.partition(' ')
+        # get block around test name
+        insub = "IN_SUBPROCESS_" + curr_test
+        for c in "/\\,:.":
+            insub = insub.replace("/", "_")
+        defined = os.environ.get(insub, None)
+        # fork process
+        if defined:
+            return f(*args, **kwargs)
+        else:
+            os.environ[insub] = '1'
+            with closed_tempfile() as stdout:
+                with open(stdout, 'w+t') as fh:
+                    rtn = subprocess.call([sys.argv[0], curr_test],
```

Comment:
`argv[0]` is `'.../pytest'` here? Should we be using `[sys.executable, '-m', 'pytest', ...]` to ensure it's run on the same Python interpreter?

Reply:
yeah that seems reasonable